iohk/iohk-ops.hs

```diff
@@ -0,0 +1,276 @@
+            Info                     -> Ops.nixops                    o c "info" []
+            -- * live deployment ops
+            CheckStatus              -> Ops.checkstatus               o c
+            Start                    -> getNodeNames'
+                                        >>= Cardano.startNodes        o c
+            Stop                     -> getNodeNames'
+                                        >>= Cardano.stopNodes         o c
+            FirewallBlock{..}        -> Cardano.firewallBlock         o c from to
+            FirewallClear            -> Cardano.firewallClear         o c
+            RunExperiment Nodes      -> getNodeNames'
+                                        >>= Cardano.runexperiment     o c
+            RunExperiment Timewarp   -> Timewarp.runexperiment        o c
+            RunExperiment x          -> die $ "RunExperiment undefined for deployment " <> showT x
+            PostExperiment           -> Cardano.postexperiment        o c
+            DumpLogs{..}
+              | Nodes        <- depl -> getNodeNames'
+                                        >>= void . Cardano.dumpLogs  o c withProf
+              | Timewarp     <- depl -> getNodeNames'
+                                        >>= void . Timewarp.dumpLogs o c withProf
+              | x            <- depl -> die $ "DumpLogs undefined for deployment " <> showT x
+            PrintDate                -> getNodeNames'
+                                        >>= Cardano.printDate        o c
+            Template{..}             -> error "impossible"
+
+
+runTemplate :: Options -> Command -> IO ()
+runTemplate o@Options{..} Template{..} = do
+  homeDir <- home
+  let bname     = fromBranch tBranch
+      branchDir = homeDir <> (fromText bname)
+  exists <- testpath branchDir
+  if exists
+    then echo $ "Using existing git clone ..."
+    else cmd o "git" ["clone", fromURL Ops.iohkNixopsURL, "-b", bname, bname]
+  cd branchDir
+  cmd o "git" (["config", "--replace-all", "receive.denyCurrentBranch", "updateInstead"])
+
+  let config = Ops.mkConfig tBranch tEnvironment tTarget tDeployments tNodeLimit
+  configFilename <- T.pack . Path.encodeString <$> Ops.writeConfig config
+
+  echo ""
+  echo $ "-- " <> (unsafeTextToLine $ configFilename) <> " is:"
+  cmd o "cat" [configFilename]
+runTemplate Options{..} _ = error "impossible"
+
+runSetCardano :: Options -> Commit -> IO ()
+runSetCardano o rev = do
+  printf ("Setting cardano-sl commit to "%s%"\n") $ fromCommit rev
+  spec <- incmd o "nix-prefetch-git" [fromURL Ops.cardanoSlURL, fromCommit rev]
+  writeFile "cardano-sl-src.json" $ T.unpack spec
+
+runSetExplorer :: Options -> Commit -> IO ()
+runSetExplorer o rev = do
+  printf ("Setting cardano-sl-explorer commit to "%s%"\n") $ fromCommit rev
+  spec <- incmd o "nix-prefetch-git" [fromURL Ops.cardanoSlExplorerURL, fromCommit rev]
+  writeFile "cardano-sl-explorer-src.json" $ T.unpack spec
+
+runMiniKeys :: IO ()
+runMiniKeys = do
```

Comment:
I don't see the purpose of `mini-keys`. In future we will use development flag not to require keys.

`datadog-api` and `datadog-application` shouldn't be ever needed since they are part of production config files.